kuksa-sdk/src/test/kotlin/org/eclipse/kuksa/connectivity/databroker/docker/DockerDatabrokerContainer.kt

```diff
@@ -0,0 +1,142 @@
+import com.github.dockerjava.core.DockerClientImpl
+import com.github.dockerjava.httpclient5.ApacheDockerHttpClient
+import org.eclipse.kuksa.connectivity.databroker.DATABROKER_TIMEOUT_SECONDS
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+private const val KEY_PROPERTY_DATABROKER_TAG = "databroker.tag"
+private const val KEY_ENV_DATABROKER_TAG = "DATABROKER_TAG"
+private const val DEFAULT_DATABROKER_TAG = "master"
+
+/**
+ * Starts and stops the Databroker Docker Container. Per default the image with the master tag is pulled and started.
+ * The version of the image can be influenced by either a System Property with value "databroker.tag" or an
+ * Environment Variable with value "DATABROKER_TAG".
+ *```
+ * Samples
+ *
+ * System Property:
+ * ./gradlew clean build -Ddatabroker.tag="0.4.1"
+ *
+ *
+ * Environment Variable:
+ * export DATABROKER_TAG="0.4.1"
+ * ./gradlew clean build
+ *
+ * or
+ *
+ * DATABROKER_TAG="0.4.1" ./gradlew clean build
+ * ```
+ */
+abstract class DockerDatabrokerContainer(
+    protected val containerName: String,
+    protected val port: Int,
+) {
+    protected open val hostConfig: HostConfig = HostConfig.newHostConfig()
+        .withNetworkMode("host")
+        .withAutoRemove(true)
+        .withPortBindings(
+            PortBinding(
+                Ports.Binding("0.0.0.0", ""),
+                ExposedPort(port, InternetProtocol.TCP),
+            ),
+        )
+
+    protected val repository = "ghcr.io/eclipse/kuksa.val/databroker"
+
+    private val databrokerTag = System.getProperty(KEY_PROPERTY_DATABROKER_TAG)
+        ?: System.getenv(KEY_ENV_DATABROKER_TAG)
+        ?: DEFAULT_DATABROKER_TAG
+
+    protected val dockerClient: DockerClient
+
+    init {
+        val config = DefaultDockerClientConfig.createDefaultConfigBuilder().build()
+
+        val timeout = Duration.ofSeconds(DATABROKER_TIMEOUT_SECONDS)
+        val dockerHttpClient = ApacheDockerHttpClient.Builder()
+            .dockerHost(config.dockerHost)
+            .sslConfig(config.sslConfig)
+            .connectionTimeout(timeout)
+            .responseTimeout(timeout)
+            .build()
+
+        dockerClient = DockerClientImpl.getInstance(config, dockerHttpClient)
+    }
+
+    fun start() {
+        stop()
+
+        pullDatabrokerImage(databrokerTag)
+        val databrokerContainer = createDatabrokerContainer(databrokerTag)
+        startContainer(databrokerContainer.id)
+    }
+
+    fun stop() {
+        val databrokerNames = listOf(containerName)
+        val dockerContainers = dockerClient.listContainersCmd()
+            .withNameFilter(databrokerNames)
+            .exec()
+
+        dockerContainers.forEach { container ->
+            try {
+                dockerClient.stopContainerCmd(container.id).exec()
+            } catch (_: NotModifiedException) {
+                // thrown when a container is already stopped
+            }
+        }
+    }
+
+    private fun pullDatabrokerImage(tag: String) {
```

Comment:
The class is already called DockerDatabrokerContainer so "pullImage" and "create" fells a bit less redundant.